src/main/java/fr/openent/viescolaire/service/impl/DefaultUserService.java

```diff
@@ -804,17 +803,20 @@ public void getTeachers(String idEtablissement, Handler<Either<String, JsonArray
     @Override
     public Future<JsonArray> getTeachersWithClassIds(String structureId) {
         Promise<JsonArray> promise = Promise.promise();
-        String query =  "MATCH (s:Structure {id: {structureId}}) " +
```

Comment:
On remonte les enseignants avec leurs classes. On cumule les données du champ `classes` avec les classes correspondants à la relation `MATCH (u: User)-[:IN]->(p:ProfileGroup)-[:DEPENDS]->(c: Class)-[:BELONGS]->(s: Structure)`

Reply:
Il faudrait vérifier où est-ce que cette méthode est utilisé (compétences, viesco, edt, cdt etc...)

Juste s'assurer qu'on ne change pas un truc "majeur"

Reply:
J'ai créé cette méthode pour l'initialisation 1D, donc aucun risque que cela soit utilisé ailleurs 😄